07.bowling_object/lib/game.rb

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative 'frame'
+
+class Game
+  attr_reader :marks
+
+  def initialize(marks)
+    @marks = marks
+  end
+
+  def score
+    frames = generate_frames
+    score = 0
+    frames.each_with_index do |frame, idx|
+      score += idx == 9 ? frame.total_score : score_nonfinal(frame, frames, idx)
+    end
+    score
+  end
+
+  private
+
+  def generate_frames
+    rolls = parse_marks
+    rolls.map { |f| Frame.new(*f[0..2]) }
```

Comment:
```suggestion
    rolls.map { |roll| Frame.new(*roll[0..2]) }
```

rollsの中身はrollだと思うので、`f`だと少し違和感がありますね。